cpp/tests/io/csv_test.cpp

```diff
@@ -2194,4 +2194,38 @@ TEST_F(CsvReaderTest, DtypesMapInvalid)
   EXPECT_THROW(cudf_io::read_csv(in_opts), cudf::logic_error);
 }
 
+TEST_F(CsvReaderTest, CsvDefaultOptionsWriteReadMatch)
+{
+  auto filepath = temp_env->get_temp_dir() + "issue.csv";
+
+  // make up some kind of dataframe
+  auto constexpr num_rows = 10;
+
+  auto int32_column       = []() {
+    auto values = random_values<int32_t>(num_rows);
+    return column_wrapper<int32_t>(values.begin(), values.end());
+  }();
+  auto int64_column = []() {
+    auto values = random_values<int64_t>(num_rows);
+    return column_wrapper<int64_t>(values.begin(), values.end());
+  }();
+  std::vector<cudf::column_view> input_columns{int32_column, int64_column};
+
+  cudf::table_view original_table{input_columns};
```

Comment:
AAA = always use auto. Generally speaking this will actually reduce the code you write, and make variable names line up better, making them easier to read.
```suggestion
  auto input_columns  = std::vector<cudf::column_view>{int32_column, int64_column};
  auto original_table = cudf::table_view{input_columns};
```
with this change, it becomes clear that `input_columns` and `original_table` are highly correlated, but their names are not highly aligned. Consider changing one or both to match.
```suggestion
  auto input_columns = std::vector<cudf::column_view>{int32_column, int64_column};
  auto input_table   = cudf::table_view{input_columns};
```